src/openssh/patch/0002-Save-remote-address-to-environment-variable-for-TACA.patch

```diff
@@ -0,0 +1,50 @@
+From 9d74466e845376040c8cf1340847025b9387f00f Mon Sep 17 00:00:00 2001
+From: liuh-80 <liuh@microsoft.com>
+Date: Tue, 11 Oct 2022 17:03:01 +0800
+Subject: [PATCH] Save remote address to environment variable for TACACS
+
+---
+ session.c             | 4 ++++
+ sshd.c                | 4 ++++
+ 2 files changed, 8 insertions(+)
+
+diff --git a/session.c b/session.c
+index a638ceef1..7a35554bb 100644
+--- a/session.c
++++ b/session.c
+@@ -618,6 +618,10 @@ do_exec_pty(struct ssh *ssh, Session *s, const char *command)
+ 
+ 		/* Close the extra descriptor for the pseudo tty. */
+ 		close(ttyfd);
++
++        /* If TACACS enabled, nss-tacplus will read remote IP from this environment variable and send to tacacs server. */
++		const char *remote_ip = ssh_remote_ipaddr(ssh);
++		setenv("SSH_REMOTE_IP", remote_ip, 1);
```

Comment:
>SSH_REMOTE_IP [](http://example.com/codeflow?start=11&length=13)

Could you explain why vanilla env var `SSH_CLIENT` could not be used in your case?

Reply:
When a remote user connect to sshd, sshd will do following steps:
1. Fork a new sshd sub-process.
2. Check user authorization.
3. Create new session and create SSH_CLIENT/SSH_CONNECTION environments.

Because the authorization happened at step2, the SSH_CONNECTION environments still not be created.
Another reason SSH_CONNECTION must be created in step3 is because SSH_CONNECTION also contains tty information, and tty is created during create session.